lte/gateway/c/oai/tasks/s1ap/s1ap_mme_encoder.c

```diff
@@ -19,437 +19,145 @@
-          (int) message_p->direction);
+          (int) pdu->present);
       break;
   }
-
-  return -1;
+  ASN_STRUCT_FREE_CONTENTS_ONLY(asn_DEF_S1ap_S1AP_PDU, pdu);
+  return ret;
 }
 
 //------------------------------------------------------------------------------
 static inline int s1ap_mme_encode_initiating(
-    s1ap_message* message_p, uint8_t** buffer, uint32_t* length) {
-  switch (message_p->procedureCode) {
-    case S1ap_ProcedureCode_id_downlinkNASTransport:
-      return s1ap_mme_encode_downlink_nas_transport(message_p, buffer, length);
+    S1ap_S1AP_PDU_t* pdu, uint8_t** buffer, uint32_t* length) {
+  asn_encode_to_new_buffer_result_t res = {NULL, {0, NULL, NULL}};
+  DevAssert(pdu != NULL);
 
+  switch (pdu->choice.initiatingMessage.procedureCode) {
+    case S1ap_ProcedureCode_id_downlinkNASTransport:
     case S1ap_ProcedureCode_id_InitialContextSetup:
-      return s1ap_mme_encode_initial_context_setup_request(
-          message_p, buffer, length);
-
     case S1ap_ProcedureCode_id_UEContextRelease:
-      return s1ap_mme_encode_ue_context_release_command(
-          message_p, buffer, length);
-
-    case S1ap_ProcedureCode_id_Paging:
-      return s1ap_mme_encode_paging(message_p, buffer, length);
-
-    case S1ap_ProcedureCode_id_UEContextModification:
-      return s1ap_mme_encode_ue_context_modification_request(
-          message_p, buffer, length);
-
     case S1ap_ProcedureCode_id_E_RABSetup:
-      return s1ap_mme_encode_e_rab_setup(message_p, buffer, length);
-
+    case S1ap_ProcedureCode_id_E_RABModify:
     case S1ap_ProcedureCode_id_E_RABRelease:
-      return s1ap_mme_encode_e_rab_release_command(message_p, buffer, length);
-
+    case S1ap_ProcedureCode_id_HandoverResourceAllocation:
+    case S1ap_ProcedureCode_id_MMEStatusTransfer:
+    case S1ap_ProcedureCode_id_Paging:
     case S1ap_ProcedureCode_id_MMEConfigurationTransfer:
-      return s1ap_mme_encode_mme_configuration_transfer(
-          message_p, buffer, length);
+      break;
 
     default:
-      OAILOG_DEBUG(
+      OAILOG_NOTICE(
           LOG_S1AP, "Unknown procedure ID (%d) for initiating message_p\n",
-          (int) message_p->procedureCode);
-      break;
+          (int) pdu->choice.initiatingMessage.procedureCode);
+      *buffer = NULL;
+      *length = 0;
+      return -1;
   }
 
-  return -1;
+  memset(&res, 0, sizeof(res));
+  res = asn_encode_to_new_buffer(
+      NULL, ATS_ALIGNED_CANONICAL_PER, &asn_DEF_S1ap_S1AP_PDU, pdu);
+  *buffer = res.buffer;
+  *length = res.result.encoded;
+  return 0;
 }
 
 //------------------------------------------------------------------------------
 static inline int s1ap_mme_encode_successfull_outcome(
-    s1ap_message* message_p, uint8_t** buffer, uint32_t* length) {
-  switch (message_p->procedureCode) {
+    S1ap_S1AP_PDU_t* pdu, uint8_t** buffer, uint32_t* length) {
+  asn_encode_to_new_buffer_result_t res = {NULL, {0, NULL, NULL}};
+  DevAssert(pdu != NULL);
```

Comment:
No need to assert here, just log an ERROR message and return. The entire process should not crash if one message is malformed.